File: C/rand64.c
#include "rand/rand64.h"

rand64_t rand64_init(long long seed)
{
	rand64_t ret;
    ret.seed = abs(seed) % 9999999L + 1L;
    ret.curr = ret.seed;
    ret.iter = 0L;
    return ret;
}

long long rand64_range(rand64_t* rand, long long min, long long max)
{
    rand->curr = (rand->curr * 125L ) % 2796203L;
	rand->iter = rand->iter + 1;
    return rand->curr % (max - min + 1L) + min;
}

int rand64_rand(rand64_t* rand, long long max)
{
    return rand64_range(rand, 0L, max);
}

File: C#/Source/Rand/Rand64.cs
using System;

namespace Rand
{
    public class Rand64
    {
        public long Seed;
        public long Current;
        public long Iterator;

        public Rand64(long Seed)
        {
            this.Seed = (Math.Abs(Seed)) % 9999999L + 1L;
            this.Current = this.Seed;
            this.Iterator = 0L;
        }

        public long Range(long Minimum, long Maximum)
        {
            this.Current = ( this.Current * 125L ) % 2796203L;
            this.Iterator = this.Iterator + 1L;
            return this.Current % (Maximum - Minimum + 1L) + Minimum;
        }

        public long Rand(long Maximum)
        {
            return this.Range(0L, Maximum);
        }
    }
}
